Weather/Views/WeatherView.swift

```diff
@@ -8,19 +8,24 @@
 
 import UIKit
 
-class WeatherView: UIView {
+protocol WeatherDelegate {
```

Comment:
Why delegate? I think you misunderstanding the MVP pattern. I think we should discuss it.